pages/DSNP/UserData.md

```diff
@@ -0,0 +1,85 @@
+# User Data
+
+DSNP stores user-related data of well defined formats associated with a user's DSNP Identifier.
+Data may be public or stored in encrypted form.
+The [Avro](https://avro.apache.org) schema specification is used to define the binary representation of relevant data types.
+
+## User Data Types
+
+DSNP implementations MUST support the following user data types:
+
+| System Name | Version | Encryption Algorithm | Compression Codec | Avro Object Type |
+| --- | --- | --- | --- | --- |
+| `publicFollows` | 1.2 | NONE | [`brotli`](https://brotli.org) | [GraphEdge](Graph.md#edges) |
+| `privateFollows` | 1.2 | `curve25519xsalsa20poly1305` |  [`brotli`](https://brotli.org) | [GraphEdge](Graph.md#edges) |
+| `privateConnections` | 1.2 | `curve25519xsalsa20poly1305` | [`brotli`](https://brotli.org) | [GraphEdge](Graph.md#edges) |
+| `privateConnectionPRIds` | 1.2 | NONE | NONE | [PRId](Graph.md#pseudonymous-relationship-identifiers) |
```

Comment:
so no compression for PRIds based on experiments?

Reply:
We need a decision on Bloom (or other type of) Filter because that seems like the only way to save significant space. Without that I feel like a <5% gain (with delta encoding) is not worth the burden it puts on rolling our own DSNP-specific compression codec.

Reply:
Decision for initial schema is to stick with an uncompressed list of PRIDs; if we find a suitable alternative approach in the future, we can bump to a new schema.

Reply:
Just a bump to highlight the suggestion for `brotli` as the codec here. It looks like there was some previous discussion of using/allowing brotli with parquetjs here: https://github.com/LibertyDSNP/parquetjs/pull/17.

Some experimentation found that the brotli codec consistently gave the best compression, though not all that much more than DEFLATE, and neither gives you too much on top of Avro's native ZigZag encoding. DEFLATE is older and widely preinstalled (e.g. multiple js libraries, built into the java.util.zip package, etc.). I don't have a strong preference.

Sample output for a list of 300 GraphEdges, assuming User Ids randomly distributed between 0 and 300M:
| Format | Size (bytes) | Total Compression |
| --- | --- | --- |
| In-memory (2 x 8-byte longs) | 4800 |
| Avro binary | 2275 | 52.6% |
| Avro+DEFLATE | 2227 | 53.6% |
| Avro+brotli | 2167 | 54.9% |

Reply:
Answer: Stick with DEFLATE